### PR TITLE
Extra labels organisation vocabs

### PR DESCRIPTION
**Pull request by vietnguyengit**

### CODE REVIEW

File: indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java [hunk 1]
```diff
@@ -88,55 +88,58 @@ this method for analysing the vocabularies of a record aka bottom-level vocabs (
...
         List<String> results = new ArrayList<>();
-        // Iterate over the top-level vocabularies
-        List<JsonNode> vocabs = switch (vocabType) {
-            case AppConstants.AODN_DISCOVERY_PARAMETER_VOCABS -> self.getParameterVocabs();
-            case AppConstants.AODN_PLATFORM_VOCABS -> self.getPlatformVocabs();
-            case AppConstants.AODN_ORGANISATION_VOCABS -> self.getOrganisationVocabs();
-            default -> new ArrayList<>();
-        };
+        if (vocabType.equals(AppConstants.AODN_ORGANISATION_VOCABS)) {
+            // TODO: the logics for mapping record's organisation vocabs are heavily customised for a manual approach, AI now or later?
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java:92]: should I do it the old school if-else, or leave here empty for Yuxuan's future magic? @utas-raymondng

File: indexer/src/main/java/au/org/aodn/esindexer/service/IndexerServiceImpl.java [hunk 2]
```diff
@@ -196,17 +196,13 @@ protected StacCollectionModel getMappedMetadataValues(String metadataValues) thr
         }
 
         // organisation vocabs
-        List<String> processedOrganisationVocabs = vocabService.extractVocabLabelsFromThemes(stacCollectionModel.getThemes(), AppConstants.AODN_ORGANISATION_VOCABS);
-        if (!processedOrganisationVocabs.isEmpty()) {
-            stacCollectionModel.getSummaries().setOrganisationVocabs(processedOrganisationVocabs);
-        }
+        // TODO: the logics for mapping record's organisation vocabs are heavily customised for a manual approach, AI now or later? need dedicated service's method
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/service/IndexerServiceImpl.java:199]: will not using `extractVocabLabelsFromThemes` method for organisation vocabs, the record's vocabs are found in different fields of the records, not "themes"

File: ardcvocabs/src/main/java/au/org/aodn/ardcvocabs/service/ArdcVocabServiceImpl.java [hunk 3]
```diff
@@ -26,9 +30,60 @@ public class ArdcVocabServiceImpl implements ArdcVocabService {
...
+        return (node) -> {
+            JsonNode labelNode = node.get(key);
+            if (labelNode != null) {
+                if (labelNode.has("_value")) {
+                    return labelNode.get("_value").asText();
+                }
+                if (labelNode instanceof TextNode) {
+                    return labelNode.asText();
+                }
+                return null;
```
Comment (utas-raymondng) [inline ardcvocabs/src/main/java/au/org/aodn/ardcvocabs/service/ArdcVocabServiceImpl.java:43]: This line no need?

**Review by utas-raymondng — CHANGES_REQUESTED** [review]
Minor comment

File: indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java (hunk 1 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/service/VocabServiceImpl.java]: How complicate the logic will be?

File: ardcvocabs/src/main/java/au/org/aodn/ardcvocabs/service/ArdcVocabServiceImpl.java (hunk 3 above)
Comment (vietnguyengit) [reply ardcvocabs/src/main/java/au/org/aodn/ardcvocabs/service/ArdcVocabServiceImpl.java:43]: I think needed, there is case that the node is just a plain text, where also case it is a json-like

E.g node = "test" vs node = {"_value": "sample"}

@utas-raymond

Comment (vietnguyengit) [reply ardcvocabs/src/main/java/au/org/aodn/ardcvocabs/service/ArdcVocabServiceImpl.java:43]: In particular, that line for textNode stuff helps to extract "displayLabel" , without it, can't get 'displayLabel' like with "prefLabel" you previous familiar with, no "_value" field.

**Review by utas-raymondng — APPROVED** [review]
LGTM

**Pull request merged**